pkg/repositories/repositories.go

```diff
@@ -57,6 +57,44 @@ func NewConfig(filename string) (Config, error) {
 	return repoConfig, nil
 }
 
+// If our first config and our second config share a repository,
+// we set the afterVersion of the first to its original version, and the
+// version of the first to the version of the second.
+func CombineConfigs(currentConfig *Config, newConfig *Config) {
+
+	// Extract new repos regardless of category
+	var newRepos []Repository
+	for _, category := range newConfig.Section.Categories {
+		newRepos = append(newRepos, category.Repos...)
+	}
+
+	// Iterate through current categories and repositories, maintaining
+	// the structure.
+	for categoryIdx := range currentConfig.Section.Categories {
+		for repoIdx, currentRepo := range currentConfig.Section.Categories[categoryIdx].
+			Repos {
+			// If two repo structs share a url, they are assumed to be the same repo.
+			// Store the newVersion as the version for the current repo.
+			if newVersion := contains(newRepos, currentRepo); newVersion != "" {
+				remappedRepo 			 := currentRepo
+				remappedRepo.AfterVersion = currentRepo.Version
+				remappedRepo.Version	  = newVersion
+
+				currentConfig.Section.Categories[categoryIdx].Repos[repoIdx] = remappedRepo
+			}
+		}
+	}
+}
+
+func contains(r []Repository, e Repository) string {
```

Comment:
`contains` is a binary type of statement but we're using it here as a value return of a field too. We probably need to disambiguate that somehow.

Reply:
Definitely, I originally returned a bool, so this name is not appropriate for this function anymore